### DISCUSSION TITLE
invalid number  detected

### DISCUSSION THREAD

**Discussion by Madhav060697** [casadi/casadi#3332] [Q&A]
I have a code framed using casadi but upon running I am getting the below Error, though the same code ran perfectly in Yalmip. I am attaching a simpler version of my code, feel free to open and run it. I would greatly appreciate it if you could solve this basic doubt.

**Error:**
using casadi.Opti/solve
Error in Opti::solve [OptiNode] at .../casadi/core/optistack.cpp:157:
.../casadi/core/optistack_internal.cpp:1000: Assertion "return_success(accept_limit)" failed:
Solver failed. You may use opti.debug.value to investigate the latest values of variables. return_status is 'Invalid_Number_Detected'

Error in test (line 79)
sol=opti.solve();
**Error**

Thank you.,
Madhav

**Sample code**
 [test.m.zip](https://github.com/casadi/casadi/files/12371134/test.m.zip)

**Comment by jgillis** [comment]
See https://github.com/casadi/casadi/wiki/FAQ:-Why-am-I-getting-%22NaN-detected%22in-my-optimization%3F

In your example, you have a division by zero at the initial guess because of `opti.subject_to(Var.Tm == Pm./Var.Wm);`. Pick a nonzero initial guess for Wm: `opti.set_initial(Var.Wm, 1);`
